ui/imports/shared/controls/CountdownPill.qml

```diff
@@ -0,0 +1,127 @@
+import QtQuick 2.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+
+import AppLayouts.Communities.controls 1.0
+
+IssuePill {
+    id: root
+
+    // request timestamp
+    required property date timestamp
```

Comment:
I think that API `timestamp` and `expirationSeconds` could be improved. Those two values are interdependent, and, if I understand correctly, for given `expirationSeconds` only one value of `timestamp` makes sense.

What is specific about those properties is that it seems unlikely to bind to them. Knowing that I would expose only methods like `setExpirationDate(date)` which could set internal values of timestamp/expiration date properly. For convenience maybe sth like `setExpirationInSeconds(seconds)` would be handy as well.

Reply:
Yeah, I wasn't sure either whether to expose them as properties, or as a setter method, since these 2 are inter-dependent. Thoughts @alexjba ?

Reply:
The downside of having an initializer function is that we can't enforce it, as with `required` props

Reply:
it could internally by default intialize to current timestamp and 0 exp seconds. It would be expired by default what is still valid state.

Reply:
We don't want an expired state by default but I get the idea :)